src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1045,13 +1041,12 @@ func (r *ClusterReconciler) doDecommissionGhostBrokers(c context.Context, vClust
 		actualBrokerIDs[id] = nil
 	}
 
-	// if the admin API shows brokers that are not assigned to existing pods, decommission them
-	for i := range adminBrokers {
-		broker := adminBrokers[i]
-		if _, ok := actualBrokerIDs[broker.NodeID]; ok {
+	for i := range health.NodesDown {
+		nID := health.NodesDown[i]
+		if _, ok := actualBrokerIDs[nID]; ok {
```

Comment:
Could we add a log line or increment a metric whenever this case is hit? If I understand the logic correctly, we should never actually hit this case, right?

Reply:
If finalizers are removed from Redpanda Pods, then I think there would be too short of a window to get annotation from Pod that would point to previous Redpanda Node ID.